phalcon/mvc/model/manager.zep

```diff
@@ -1805,4 +1810,459 @@ class Manager implements ManagerInterface, InjectionAwareInterface, EventsAwareI
+		 */
+		let writeConnection = model->getWriteConnection();
+
+		/**
+		 * Fire the start event
+		 */
+		model->fireEvent("prepareSave");
+
+		/**
+		 * Save related records in belongsTo relationships
+		 */
+		let related = model->_related;
+		if typeof related == "array" {
+			if model->_preSaveRelatedRecords(writeConnection, related) === false {
+				return false;
+			}
+		}
+
+		let schema = model->getSchema(),
+			source = model->getSource();
+
+		if schema {
+			let table = [schema, source];
+		} else {
+			let table = source;
+		}
+
+		/**
+		 * Create/Get the current database connection
+		 */
+		let readConnection = model->getReadConnection();
+
+		/**
+		 * We need to check if the record exists
+		 */
+		let exists = model->_exists(metaData, readConnection, table);
+
+		if exists {
+			let model->_operationMade = Model::OP_UPDATE;
+		} else {
+			let model->_operationMade = Model::OP_CREATE;
+		}
+
+		/**
+		 * Clean the messages
+		 */
+		let model->_errorMessages = [];
+
+		/**
+		 * Query the identity field
+		 */
+		let identityField = metaData->getIdentityField(model);
+
+		/**
+		 * _preSave() makes all the validations
+		 */
+		if model->_preSave(metaData, exists, identityField) === false {
+
+			/**
+			 * Rollback the current transaction if there was validation errors
+			 */
+			if typeof related == "array" {
+				writeConnection->rollback(false);
+			}
+
+			/**
+			 * Throw exceptions on failed saves?
+			 */
+			if globals_get("orm.exception_on_failed_save") {
+				/**
+				 * Launch a Phalcon\Mvc\Model\ValidationFailed to notify that the save failed
+				 */
+				throw new ValidationFailed(model, model->getMessages());
+			}
+
+			return false;
+		}
+
+		/**
+		 * Depending if the record exists we do an update or an insert operation
+		 */
+		if exists {
+			let success = model->_doLowUpdate(metaData, writeConnection, table);
+		} else {
+			let success = model->_doLowInsert(metaData, writeConnection, table, identityField);
+		}
+
+		/**
+		 * Change the dirty state to persistent
+		 */
+		if success {
+			model->setDirtyState(
+				Model::DIRTY_STATE_PERSISTENT
+			);
+		}
+
+		if typeof related == "array" {
+
+			/**
+			 * Rollbacks the implicit transaction if the master save has failed
+			 */
+			if success === false {
+				writeConnection->rollback(false);
+			} else {
+				/**
+				 * Save the post-related records
+				 */
+				let success = model->_postSaveRelatedRecords(writeConnection, related);
+			}
+		}
+
+		/**
+		 * _postSave() invokes after* events if the operation was successful
+		 */
+		if globals_get("orm.events") {
+			let success = this->_postSave(model, success, exists);
+		}
+
+		if success === false {
+			model->_cancelOperation();
+		} else {
+			model->fireEvent("afterSave");
+		}
+
+		return success;
+	}
+
+	/**
+	 * Executes internal events after save a record
+	 */
+	protected function _postSave(<ModelInterface model, boolean success, boolean exists) -> boolean
```

Comment:
But UnitOfWork is effectively a transaction - it only makes sense to use it if multiple models are being saved at once. I want to get all the pre/post protected methods into Manager just for now so they're all together and we can move again them later.